storage/innobase/handler/handler0alter.cc

```diff
@@ -8811,6 +8868,11 @@ commit_try_norebuild(
 		DBUG_RETURN(true);
 	}
 
+	if (ha_alter_info->handler_flags & ALTER_COLUMN_UNVERSIONED) {
+		innobase_change_fields_versioning(ha_alter_info, ctx,
+						  old_table);
+	}
+
```

Comment:
After this point, `commit_try_norebuild()` can still fail, and the operation may be rolled back.
I think that the data dictionary cache adjustment should be done in `commit_cache_norebuild()`. Before commit, in `commit_try_norebuild()`, we must update the `SYS_COLUMNS` only, similar to how `innobase_rename_columns_try()` and `innobase_enlarge_columns_try()` work.